### PR TITLE
Upgrade to PETSc-3.13.3

### PR DESCRIPTION
**Pull request by fdkong**
Take an attempt to upgrade to PETSc-3.13.3

Closes ##15589

### CODE REVIEW

**Comment by fdkong** [comment]
FYI @gambka

**Review by lindsayad — DISMISSED** [review]
Remind me the reason that you have to increment the build number for libmesh? Also do you still mean to have the “do not merge” label? Very nice to see you didn’t have to do any app fix ups!! (Or maybe you did some stuff behind the scenes)

**Comment by fdkong** [comment]
> Remind me the reason that you have to increment the build number for libmesh? 

Libmesh won't pick up new petsc if we do not trigger it by changing the build number or something else.

That is, `conda-libmesh`  will most likely keep using petsc-3.12.5 even though PETSc is updated. 

@milljm  can confirm if these statements still make sense.

>Also do you still mean to have the “do not merge” label? Very nice to see you didn’t have to do any app fix ups!! (Or maybe you did some stuff behind the scenes)

Did a bit a while ago.  PETSc upgrade seems to be  very smooth this time. It is a gift for me!

**Comment by milljm** [comment]
Correct. However, this PR is incorrect, as it does not change the conda_build_config file for moose-libmesh:

 `conda/libmesh/conda_build_config.yaml`:
```yaml
<trimmed>

moose_petsc:
 - 3.12.5 build_2      <------ should be 3.13.3 build_0

<trimmed>
```

Otherwise end-users, when installing moose-libmesh, will still get petsc 3.12.5 build_2.

This no longer matters though. Because that combination works (has been for a while), and Civet, is building libMesh on it's own everywhere (remember that multi-libmesh update fiasco that blew Civet apart?).

Merge now and create a moose-libmesh changing PR later, or make the above change. Your call!

**Comment by fdkong** [comment]
> Merge now and create a moose-libmesh changing PR later, or make the above change. Your call!

OK, let us get everything right in this PR. Updated. 

Thanks so much!

File: conda/libmesh/meta.yaml
Comment (lindsayad) [inline conda/libmesh/meta.yaml:1]: Shouldn't this be a 0 now?

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: conda/libmesh/meta.yaml
Comment (milljm) [reply conda/libmesh/meta.yaml:1]: He does... but IMHO, the better thing to do, so to bring this to 0, and adjust the version to match today's date. Either will work. Perhaps, we should only increment the version when there is a change to libMesh instead?

It's good question, so we are all on the same page moving forwards.

Comment (milljm) [reply conda/libmesh/meta.yaml:1]: You have to do something with build or version for sure. Otherwise, we will overwrite the moose-libmesh package being served on mooseframework.org.

Comment (milljm) [reply conda/libmesh/meta.yaml:1]: To further complicate things, any additional changes to conda/libmesh now, will result in something odd: two moose-libmesh's being served up on https://rod.inl.gov/conda/moose/pr-15587/ (click on linux-64 or osx-64 to see what I mean).

And at that point, an ADDITIONAL change will then be required to conda/petsc, so as to get the right version of moose-libmesh.

**All I am trying to say, is the conda_build_config.yaml file's in all our conda recipes need to be pointing at the correct targets.**

Comment (lindsayad) [reply conda/libmesh/meta.yaml:1]: Yea maybe build is better to increment here, because the libmesh version hasn't changed. This is just a new "build" because it is built and linked with a different PETSc :+1:

Comment (fdkong) [reply conda/libmesh/meta.yaml:1]: Yes, `build number` makes more sense than `libmesh version`.

Comment (milljm) [reply conda/libmesh/meta.yaml:1]: yep, I also agree. I almost edited my post about 'IMHO' after I re-read what I had wrote.

Changing the build number makes more sense. So, this PR is now in good shape.

**Comment by fdkong** [comment]
Could you approve/review @milljm @lindsayad ?

**Review by lindsayad — APPROVED** [review]

**Review by milljm — APPROVED** [review]
👍

**Pull request merged**
